spark/agent/src/test/scala/za/co/absa/spline/harvester/AttributeDependencySolverSpec.scala

```diff
@@ -0,0 +1,453 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.harvester
+
+import java.util.UUID
+
+import org.scalatest.{FlatSpec, Matchers}
+import za.co.absa.spline.harvester.AttributeDependencySolver.resolveDependencies
+import za.co.absa.spline.harvester.ModelConstants.{OperationExtras, OperationParams}
+import za.co.absa.spline.model.Attribute
+import za.co.absa.spline.model.expr.{AttrRef, Binary, Expression, Generic, Literal}
+import za.co.absa.spline.producer.model.{DataOperation, ReadOperation, WriteOperation}
+
+class AttributeDependencySolverSpec extends FlatSpec with Matchers {
+
+  private val A, B, C, D, E, F, G = UUID.randomUUID()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Field name 'F' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://app.codacy.com/app/ABSA_OSS/spline/pullRequest?prid=4630741)